### ISSUE TITLE
Wrong table name for ABI when `eosio::singleton` used

### ISSUE THREAD

**Issue by Mattijah** [AntelopeIO/cdt#93]
The following ticket (bug) has been closed despite it not even being resolved and it can still be reproduced in the latest `3.2.0`. The bug completely prevents from uploading a contract. More info: https://github.com/EOSIO/eosio.cdt/issues/280

**Comment by stephenpdeos** [comment]
@Mattijah can you clarify the version number of CDT you are encountering this? Could you also include an example?

**Comment by Mattijah** [comment]
- The version number is mentioned in my comment above (`3.2.0-dev`), but since it wasn't fixed since 1.8 it probably is not working in any version.
- Example can be found in the link attached. Copy:

```c++
class [[eosio::contract("test")]] test : public contract {
    public:
        using contract::contract;
        
    [[eosio::action]]
        void whatever() {};

    struct [[eosio::table]] tbl_config {
        uint64_t y;
        uint64_t x;
    };
    
    typedef eosio::singleton<"config"_n, tbl_config> config;
};
```

Will produce:
```json
"tables": [
        {
            "name": "tbl_config", // <<< wrong
            "type": "tbl_config",
            "index_type": "i64",
            "key_names": [],
            "key_types": []
        }
```

**Comment by stephenpdeos** [comment]
@Mattijah 3.2.0 is prob the leap version. Most recent cdt version is 3.1.0. Thanks for the example.
